src/liballoc/lib.rs

```diff
@@ -69,7 +69,10 @@
        issue_tracker_base_url = "https://github.com/rust-lang/rust/issues/",
        test(no_crate_inject, attr(allow(unused_variables), deny(warnings))))]
 #![no_std]
-#![needs_allocator]
+
+// Don't inject an allocator in stage0 since we explicitly use alloc_system, jemalloc
+// being possibly unavailable or disallowed
+#![cfg_attr(not(stage0), needs_allocator)]
```

Comment:
That's the part where I'm not crystal clear what's happening; I had to reinstate this line to avoid the issue described in the issue, with make install starting from stage0 after a successful build